src/components/ShareDialog.jsx

```diff
@@ -36,51 +37,52 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-export default function ShareDialog(props) {
+export default function ShareDialog() {
+  const location = useLocation();
+  const { showShareDialog, shareTitle, shareURL } = location.state || {
+    showShareDialog: false,
+  };
+
+  const history = useHistory();
+  const closeDialog = () =>
+    history.replace({ state: { showShareDialog: false } });
```

Comment:
On Render, there's a slight glitch when you dismiss the ShareDialog. I guess maybe you should keep the title/link state but only change `show` to `false` so that when the dialog fades the glitch doesn't show up.